tests/Collection/CollectionFunctionalTest.php

```diff
@@ -311,6 +311,71 @@ public function testMapReduce()
         $this->assertNotEmpty($result->getCounts());
     }
 
+    public function testDontInheritWriteConcernInTransaction()
+    {
+        $this->skipIfTransactionsAreNotSupported();
+
+        $this->createCollection();
+
+        $session = $this->manager->startSession();
+        $session->startTransaction([
+            'readConcern' => new \MongoDB\Driver\ReadConcern("snapshot"),
+            'writeConcern' => new \MongoDB\Driver\WriteConcern(\MongoDB\Driver\WriteConcern::MAJORITY)
+        ]);
+
+        $collection = new Collection(
+            $this->manager,
+            $this->getDatabaseName(),
+            $this->getCollectionName(),
+            [ 'writeConcern' => new \MongoDB\Driver\WriteConcern(1) ]
+        );
+
+        (new CommandObserver)->observe(
+            function() use ($collection, $session) {
+                $collection->insertOne(
+                    ['test' => 'foo'],
+                    [
+                        'session' => $session,
+                    ]
+                );
+            },
+            function(array $event) {
+                $this->assertObjectNotHasAttribute('writeConcern', $event['started']->getCommand());
+            }
+        );
+
+        $this->dropCollection();
```

Comment:
IIRC, we only need to explicitly _create_ collections for transaction tests. `MongoDB\Tests\Collection::tearDown()` already drops the collection under test iff the test has not failed (otherwise it leaves it in place for inspection), so this can be removed.